pkg/proxy/ipvs/proxier.go

```diff
@@ -1905,13 +1905,13 @@ func (proxier *Proxier) createAndLinkeKubeChain() {
 			if chain, ok := existingNATChains[ch.chain]; ok {
 				writeBytesLine(proxier.natChains, chain)
 			} else {
-				writeLine(proxier.natChains, utiliptables.MakeChainLine(kubePostroutingChain))
+				writeLine(proxier.natChains, utiliptables.MakeChainLine(ch.chain))
```

Comment:
This definitely looks right, but trying to dig up history as to why this was pointing to KUBE-POSTROUTING. There may have been some defaulting that is no longer relevant.

Reply:
I think iptables has the same problem last time I checked

Reply:
Nvm, looks like it was fixed: 

https://github.com/kubernetes/kubernetes/blob/d29665cc17f05fbb66ea4b77d2fefbf39abd4e82/pkg/proxy/iptables/proxier.go#L931-L946

Reply:
I guess it was introduced when reducing duplicated code in this PR: https://github.com/kubernetes/kubernetes/commit/339f92c175932d46bd0ef64b419948dbb97b7feb#diff-1f2043db3a45960023376d03e4dd43762d1382ad7ea94e3f26c956410b749c97L729.